### PR TITLE
refactor + job management

### PR DESCRIPTION
**Pull request by phiros**
My refactor of core. I swapped out seneca with EventEmitter3 (don't really care if we use EventEmitter2 or 3). For persistence I went with lokijs which I encapsulated in a class (using ES6 classes). We can change this latter on. 

BTW: iojs seems to dislike Primus. On ubuntu 15.04  with the most recent version of iojs (installed with `nvm`) I got Segmentation faults...
(could also be the combination Primus + traceur-runner + iojs). 
However, `npm start` and `npm test` run flawlessly in node v0.12.7.

### CODE REVIEW

File: lib/models/drone.js [hunk 1]
```diff
@@ -1,17 +1,35 @@
...
 
-exports.checkIn = function (id, data) {
-  return Promise.resolve({ droneId: id, onPostAuth: true, status: 'ready' })
+class droneCollection extends SimpleCollection {
+  constructor (schema) {
+    super(schema)
+  }
 }
+exports.Drone = Drone
+exports.DroneSchma = DroneSchema
```
Comment (knownasilya) [inline lib/models/drone.js]: Typo?

File: lib/models/drone.js [hunk 2]
```diff
@@ -1,17 +1,35 @@
...
-exports.create = function (data) {
-  data.id = 1
-  data.token = 'mocktoken.1'
+let DroneSchema = {
+  'title': 'Generic Drone Schema',
+  'type': 'object',
+  'properties': {
+    'id': { 'type': 'string' }
+  }
+}
```
Comment (knownasilya) [inline lib/models/drone.js:11]: There's no really reason to put these properties in quotes, right?

Comment (phiros) [inline lib/models/drone.js:11]: Don't know. On http://json-schema.org/ and similar sites the properties are always in quotes....

File: lib/models/drone.js (hunk 1 above)
Comment (phiros) [inline lib/models/drone.js]: fixed

**Comment by knownasilya** [comment]
Why use traceur with iojs? IOjs has ES6 features without the need for a transpiler (I'd go with babel anyways if it was needed), which explains why you were having issues.

**Comment by phiros** [comment]
hmm not sure that traceur is the problem here:

```
$ nvm use iojs
$ node --version                                                                                                             
v3.1.0
$ node index.js server                                                                                                     
[1]    15112 segmentation fault (core dumped)  node index.js server
```

Also support for ES6 features is still spotty in iojs and sometimes even a little bit buggy (see: http://kangax.github.io/compat-table/es6/)

**Comment by knownasilya** [comment]
Why `node index.js server` and not just `node index.js`?

**Comment by knownasilya** [comment]
> Also support for ES6 features is still spotty in iojs and sometimes even a little bit buggy

Then node + babel would probably be best.

**Comment by phiros** [comment]
`node index.js server`
This is just to distinguish between running index.js as a server and including the code of index for the tests. The check you used doesn't work if you run index.js trough `traceur-runner` or something other than `node`.

Regarding ES6:
I just went with traceur because it just worked on my machine. For some reason I couldn't get babel to work properly...
If you want you can switch the code over to babel (I guess you only need to change package.json?; I've never used babel before...).

**Pull request merged**